From 2f34315634e32a9c2ba77f95f2f893d3011ad99a Mon Sep 17 00:00:00 2001
From: Joachim Wiberg <troglobit@gmail.com>
Date: Mon, 14 Jun 2021 11:18:18 +0200
Subject: [PATCH] fnread(): make sure to NUL terminate buffer

Note, this is dead code, currently unsused in Finit.  Possibly an
external plugin makes use of it, but even that is highly unlikely.

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 src/util.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/util.c b/src/util.c
index 84eab60..959581b 100644
--- a/src/util.c
+++ b/src/util.c
@@ -177,7 +177,8 @@ int fnread(char *buf, size_t len, char *fmt, ...)
 	if (!fp)
 		return -1;
 
-	len = fread(buf, sizeof(char), len, fp);
+	len = fread(buf, sizeof(char), len - 1, fp);
+	buf[len] = 0;
 	fclose(fp);
 
 	return (int)len;
@@ -216,7 +217,7 @@ int fnwrite(char *value, char *fmt, ...)
 
 int fngetint(char *path, int *val)
 {
-	char buf[10];
+	char buf[64];
 
 	if (fnread(buf, sizeof(buf), path) <= 0)
 		return -1;
